common/locale.go

```diff
@@ -0,0 +1,38 @@
+package common
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+// LocaleCode the locale code to use to determine the language to use
+const LocaleCode = ContextKey("LocaleCode")
+
+// SetLocaleCode sets the Locale code used to set the language
+func SetLocaleCode(req *http.Request) context.Context {
+	lang := ExtractLangFromSubDomain(req)
+
+	if c, err := req.Cookie("lang"); err == nil && len(c.Value) > 0 {
+		lang = ExtractLangFromCookie(c)
+	}
+
+	return context.WithValue(req.Context(), LocaleCode, lang)
```

Comment:
I would also consider pulling this out into it's own function - as there could be cases where we might want to use this directly.

```
// using the Lang type I mentioned in my earlier comment
func AddLangToContext(ctx context.Context, lang Lang) context.Context {
...
}

Reply:
After further investigation we encountered an issue; the context was not passed in the `allRouter` handler due to being an 'http' request, not a 'rchttp' request. As such we have instead added this to a header and this line is now `req.Header.Set("LocaleCode", localeCode)`